File: c++/Easy/Part004/Characters at Multiple of X.cpp
#include <iostream>
#include <string>

using namespace std;

void characters_at_multiples_of_x(string S, int X) {
    string result = "";
    
    // Iterate over the string S
    for (int i = X - 1; i < S.length(); i += X) {
        result += S[i];
    }
    
    // Print the result
    cout << result << endl;
}

int main() {
    string S;
    int X;
    
    // Read input values
    getline(cin, S);
    cin >> X;
    
    // Call the function to calculate and print characters at multiples of X
    characters_at_multiples_of_x(S, X);
    
    return 0;
}

File: Java/Easy/Part004/Characters at Multiple of X.java
import java.util.Scanner;

public class Hello {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read the string S
        String S = scanner.nextLine();

        // Read the integer X
        int X = scanner.nextInt();

        // Initialize an empty result string
        StringBuilder result = new StringBuilder();

        // Loop through the string and collect characters at positions which are multiples of X
        for (int i = X - 1; i < S.length(); i += X) {
            result.append(S.charAt(i));
        }

        // Print the result
        System.out.println(result.toString());
    }
}
